rfcs/0017-tx-valid-since/0017-tx-valid-since.md

```diff
@@ -2,334 +2,191 @@
-                    }
-                }
-            }
-            Ok(())
-        }
-        
-        pub fn verify(&self) -> Result<(), Error> {
-            for (cell_meta, input) in self
-                .rtx
-                .resolved_inputs
-                .iter()
-                .zip(self.rtx.transaction.inputs()) {
-                // ignore empty since
-                let since: u64 = input.since().unpack();
-                if since == 0 {
-                    continue;
-                }
-                let since = Since(since);
-                // check remain flags
-                if !since.flags_is_valid() {
-                    return Err(TransactionError::InvalidSince).into());
-                }
-                
-                // verify time lock
-                self.verify_absolute_lock(since)?;
-                self.verify_relative_lock(since, cell_meta)?;
-            }
-            Ok(())
-        }
-    }
-```
+Transaction input adds a new field `since`, which prevents the transaction to be mined before an absolute or relative time.
+
+There are two important moments for a cell, the time of creation and destruction. The cell is created when the transaction producing it as an output has been committed into a block in the chain. And the destruction time is when the transaction consuming the cell as an input has been committed.
+
+The field `since` specifies the minimal destruction time of the input cell. The input is said premature when the `since` field is present and the minimal destruction time has not arrived yet. A transaction is premature when any of its input is premature. The block should not commit any premature transactions.
+
+![](cell-lifecycle.jpg)
+
+## Specification
+
+### Encoding
+
+The `since` is a `u64` (unsigned 64-bit integer) type field in the transaction input. The highest 8 bits of the `since` field is the `flags`. The remaining `56` bits represent the `value`.
+
+![](since-encoding.jpg)
+
+* The highest bit is the relative flag.
+    * `0`: The `value` is absolute that the cell destruction time should not be earlier than `value`.
+    * `1`: The `value` is relative that the cell destruction time should not be earlier than cell creation time plus `value`.
+* The following two bits choose the time metric used to represent the creation time, destruction time, and the `value`.
+    * `00`: Use block number.
+    * `01`: Use epoch number with fraction.
+    * `10`: Use the timestamp median of previous block headers.
+    * `11`: Invalid. Transaction should not set metric flag to `11`.
+* The next 5 bits are reserved for future extension. They must be set to all zeros now.
+
+### Since Verification
+
+The consensus iterates each input to check whether the cell is mature:
+
+* The cell is mature when the `since` is 0.
+* If the relative flag is `0` (absolute), the cell is mature when the destruction time is larger than or equal to the `value`.
```

Comment:
I think using "destruction time" to define validation rules is inappropriate, because "destruction time" can only be determined when the transaction is included in a block, but the validation here happens before the transaction is included.